utils/utils_test.go

```diff
@@ -1,27 +1,72 @@
 package utils
 
 import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"runtime"
 	"testing"
 	"time"
 )
 
-func TestUtilIntegration(t *testing.T) {
+func TestEOFKeySequenceText(t *testing.T) {
+	current := eofKeySequenceText()
+
+	if runtime.GOOS == "windows" {
+		if current != "Ctrl+Z and Enter" {
+			t.Errorf("Unexpected EOF key sequence text. Wanted: Ctrl+Z and Enter  Got: %v", current)
+		}
+	}
+
+	if current != "Ctrl+D" {
+		t.Errorf("Unexpected EOF key sequence text. Wanted: Ctrl+D Got: %v", current)
+	}
+}
+
+// Test the complex function of the AddSecret helper method in a temporary path to ensure
+// we have validated the "from scratch" use case.
+func TestAddSecretHelper(t *testing.T) {
+
+	// Test New Secret in New Context
+	tempPath, tempPathErr := ioutil.TempDir("", "TestAddSecretHelper")
+	currPath, _ := os.Getwd()
+	t.Logf("Using temporary path %v", tempPath)
+	if tempPathErr != nil {
+		t.Skipf("Failed to create temporary path with error: %v", tempPathErr)
+	}
+	defer os.RemoveAll(tempPath)
+	os.Chdir(tempPath)
+
+	currWD, _ := os.Getwd()
+	t.Logf("In current path %v", currWD)
 	hush := Secret{
 		Name:       "TEST",
 		Cyphertext: "TESTCASEADDSECRET",
 		Created:    time.Now(),
 		Encoding:   "plain",
 	}
 
-	AddSecret(hush)
-	state := ReadSecrets()
+	AddSecret(hush, "", true)
+	state, _, _ := ReadSecrets()
```

Comment:
shouldn't we be testing the other returned state as well?

Reply:
I'm not sure i follow along with this comment, could you clarify? I'm happy to update the test.

Reply:
ah, yes. what I meant was that `ReadSecrets` returns `(Secrets, string, error)`, but we're only checking the `Secrets` here. I just meant that we should add checks to make sure the string and error portion of the return are coming back as expected

Reply:
I thought so, but wanted to make sure :+1:  Thank you :)